stubstub/modelize/builders.ts

```diff
@@ -0,0 +1,281 @@
+import uuid from 'uuid';
+import SHA3 from 'crypto-js/sha3';
+import encHex from 'crypto-js/enc-hex';
+
+import { createCollectionPub } from 'server/collectionPub/queries';
+import {
+	ActivityItem,
+	Community,
+	FacetBinding,
+	Member,
+	Release,
+	SubmissionWorkflow,
+	User,
+	UserSubscription,
+} from 'server/models';
+import { createPub } from 'server/pub/queries';
+import { createCollection } from 'server/collection/queries';
+import { createDoc } from 'server/doc/queries';
+import { createPage } from 'server/page/queries';
+import { createCommunity } from 'server/community/queries';
+import { getEmptyDoc } from 'client/components/Editor';
+import {
+	ActivityItem as ActivityItemType,
+	Collection as CollectionType,
+	Community as CommunityType,
+	Member as MemberType,
+	UserWithPrivateFields as UserType,
+	Release as ReleaseType,
+	Pub as PubType,
+	FacetBinding as FacetBindingType,
+	SubmissionWorkflow as SubmissionWorkflowType,
+	UserSubscription as UserSubscriptionType,
+} from 'types';
+
+type WithOptionalBase<
+	T extends Record<string, any>,
+	K extends keyof T | undefined = undefined,
+> = K extends keyof T ? Omit<T, K> & Partial<T> : T;
+
+/**
+ * Allows properties which are optionally null to also be optionally undefined,
+ * and allows you to specify extra optional properties.
+ *
+ * @example
+ * ```ts
+ * type MyType = { a: string; b: number; c: boolean | null; d: string | null; };
+ *
+ * type MyTypeWithNullAsAlsoUndefined = WithOptional<MyType>;
+ * // { a: string; b: number; c: boolean | null | undefined; d: string | null | undefined; }
+ *
+ * type MyTypeWithNullAsAlsoUndefinedAndOptionalA = WithOptional<MyType, 'd'>;
+ * // { a: string | undefined; b: number; c: boolean | null | undefined; d: string | null | undefined; }
+ * ```
+ *
+ */
+type WithOptional<
+	/**
+	 * The object type to operate on
+	 */
+	T extends Record<string, any>,
+	/**
+	 * @optional The keys to make optional
+	 */
+	K extends keyof T | undefined = undefined,
+> = {
+	[PossiblyNullKey in keyof WithOptionalBase<T, K>]: WithOptionalBase<
+		T,
+		K
+	>[PossiblyNullKey] extends null
+		? WithOptionalBase<T, K>[PossiblyNullKey] | undefined
+		: WithOptionalBase<T, K>[PossiblyNullKey];
+};
+
+export const builders = {
```

Comment:
main change here is that this is from

```ts

const builders = {}

builders.User = ...
```

to 

```ts

const builders = {

User: //

}
```

for better type inference

Reply:
Awesome!